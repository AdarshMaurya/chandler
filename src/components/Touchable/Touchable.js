import React from 'react';
import {
  TouchableOpacity,
  TouchableNativeFeedback,
  Platform,
  ViewPropTypes,
  View,
} from 'react-native';
import PropTypes from 'prop-types';

export const Touchable = ({style, onPress, children}) =>
  Platform.OS === 'ios' ? (
    <TouchableOpacity style={style} onPress={onPress}>
      {children}
    </TouchableOpacity>
  ) : (
    <TouchableNativeFeedback onPress={onPress}>
      <View style={style}>{children}</View>
    </TouchableNativeFeedback>
  );

Touchable.propTypes = {
  onPress: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  style: ViewPropTypes.style,
};
