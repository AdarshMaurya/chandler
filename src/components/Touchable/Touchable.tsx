import React, {PureComponent} from 'react';
import {
  Platform,
  TouchableNativeFeedback,
  TouchableOpacity,
  View,
  ViewStyle,
} from 'react-native';

interface ITouchableProps {
  style?: ViewStyle;
  onPress: (...args) => any;
  children;
}

export class Touchable extends PureComponent<ITouchableProps, {}> {
  public render() {
    const {style, onPress, children} = this.props;
    return Platform.OS === 'ios' ? (
      <TouchableOpacity style={style} onPress={onPress}>
        {children}
      </TouchableOpacity>
    ) : (
      <TouchableNativeFeedback onPress={onPress}>
        <View style={style}>{children}</View>
      </TouchableNativeFeedback>
    );
  }
}
