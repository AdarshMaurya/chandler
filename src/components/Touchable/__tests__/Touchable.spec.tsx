import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {View} from 'react-native';
import {Touchable} from '../Touchable';

describe('Components', () => {
  describe('Touchable', () => {
    it('ios should renders correctly', () => {
      const onPress = jest.fn();
      expect(
        toJson(
          shallow(
            <Touchable onPress={onPress}>
              <View />
              <View />
            </Touchable>
          )
        )
      ).toMatchSnapshot();
    });

    it('android should renders correctly', () => {
      const onPress = jest.fn();
      jest.mock('Platform', () => ({
        OS: 'android',
      }));

      expect(
        toJson(
          shallow(
            <Touchable onPress={onPress}>
              <View />
            </Touchable>
          )
        )
      ).toMatchSnapshot();
    });

    it('should call onPress when taping the list row', () => {
      const onPress = jest.fn();
      const button = shallow(
        <Touchable onPress={onPress}>
          <View />
        </Touchable>
      );
      button.prop('onPress')();
      expect(onPress).toHaveBeenCalledTimes(1);
      expect(onPress).toHaveBeenCalledWith();
    });
  });
});
