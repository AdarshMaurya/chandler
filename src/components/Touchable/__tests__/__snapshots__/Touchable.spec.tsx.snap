// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Touchable android should renders correctly 1`] = `
<DummyTouchableNativeFeedback
  onPress={[MockFunction]}
>
  <Component>
    <Component />
  </Component>
</DummyTouchableNativeFeedback>
`;

exports[`Components Touchable ios should renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
>
  <Component />
  <Component />
</TouchableOpacity>
`;
